clang/test/SemaSYCL/sycl-device-intel-fpga-local.cpp

```diff
@@ -0,0 +1,24 @@
+// RUN: %clang_cc1 -fsycl -fsycl-is-device -fsyntax-only -ast-dump -verify -pedantic %s | FileCheck %s
+
+// Test that checks global constant variable (which allows the redeclaration) since
+// IntelFPGAConstVar is one of the subjects listed for [[intel::max_replicates()]] attribute.
+
+// Merging of different arg values.
+//CHECK: VarDecl{{.*}}var_max_replicates
```

Comment:
Can you remove this AST from diagnostic and add AST check for a valid variable instead?